http_proxy_test.go

```diff
@@ -109,3 +109,37 @@ func TestNopDialer(t *testing.T) {
 		t.Fatalf("expected %v, got %v", nopDialerErr, err)
 	}
 }
+
+func TestIsLocalhost(t *testing.T) {
+	cfg := DefaultHTTPProxyConfig()
+	p, err := NewHTTPProxy(cfg, nil, nil, nil, stdlog.Default())
```

Comment:
May we have a test for a handler mode too (`newHTTPProxy`)?

Reply:
That suite should work without reading hosts file.

Reply:
We can do that in a separate issue. It uses convention in this file.